src/app/screens/Transactions/index.tsx

```diff
@@ -0,0 +1,61 @@
+import Container from "@components/Container";
+import TransactionsTable from "@components/TransactionsTable";
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { toast } from "react-toastify";
+import { useSettings } from "~/app/context/SettingsContext";
+import { convertPaymentsToTransactions } from "~/app/utils/payments";
+import msg from "~/common/lib/msg";
+import { Payment, Transaction } from "~/types";
+
+function Transactions() {
+  const { t } = useTranslation("translation", {
+    keyPrefix: "transactions",
+  });
+
+  const { settings, getFormattedFiat } = useSettings();
+  const [payments, setPayments] = useState<Transaction[]>([]);
+
+  const fetchData = useCallback(async () => {
+    try {
+      const { payments } = await msg.request<{ payments: Payment[] }>(
+        "getPayments"
+      );
+      const transactions: Transaction[] =
+        convertPaymentsToTransactions(payments);
+
+      for (const transaction of transactions) {
+        transaction.totalAmountFiat = settings.showFiat
+          ? await getFormattedFiat(transaction.totalAmount)
+          : "";
+      }
+
+      setPayments(transactions);
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error) toast.error(`Error: ${e.message}`);
+    }
+  }, [settings, getFormattedFiat]);
+
+  useEffect(() => {
```

Comment:
This might be a good candidate for using SWR too, if we can get that other PR merged. Maybe could you add a TODO comment if you agree?

Reply:
I thought about it. But I wasn't too sure how and when to reset/mutate the cache. I added a Todo comment.